feat: The Graph migration (moving on from GoldSky)

wsdt opened the pull request:
All subgraphs deployed. If approved then will release new major version and propose that we extensively test the gateway on staging and deploy lightbridge for testnet with the update. 

Make sure to set `THE_GRAPH_API_KEY` in your .env when using this package in your app.

wsdt committed:
fix: graph uris

wsdt committed:
fix: replaced managable uris

wsdt committed:
fix: migration

wsdt committed:
feat: boba graphs

wsdt committed:
fix: bnb services

wsdt committed:
fix: env-key

wsdt committed:
fix: dotenv

wsdt committed:
fix: retain old data structure, tests for graphs

wsdt committed:
fix: fix bnb graph link; update badge coverage; dotenv import

wsdt committed:
fix: dns names for bnb services

wsdt committed:
fix: rm type module

wsdt committed:
fix: rm logs, es6 config

wsdt committed:
fix: anchorage boba bnb

boyuan-chen commented:
@wsdt Could you please solve the conflicts?

wsdt committed:
fix: conflicts and timestamp check

sk-enya commented on src/graphql.service.ts:
```diff
@@ -5,115 +5,128 @@ import {
             },
         },
         // BNB testnet
         97: {
```
can we have explorer for BNB testnet `AnchorageBridge` so we can query the proven transactions and finalized

wsdt committed:
feat: boba-sepolia graphs

wsdt committed:
fix: remaining graphs

sk-enya reviewed (approved):
Looks good!

wsdt commented:
will test all this today with the Gateway / LightBridge first.

wsdt committed:
fix: chapel

wsdt committed:
fix: updated subgraph id

boyuan-chen commented on src/graphql.service.ts:
```diff
@@ -5,115 +5,136 @@ import {
 
-
 export class GraphQLService {
+    private readonly apikey = process.env.THE_GRAPH_API_KEY ?? '815ddde7e44ffb91132567e2c4e7c558'
```
Is this the API key?

wsdt committed:
fix: add new chapel addr after ownership transfer

wsdt committed:
feat: dispute-game-factory; health-check tests

wsdt committed:
feat: dispute game factory subgraph

sk-enya committed:
change:
- bundled the packages together along with key
- updated graphq API Key name and updated readme with usage.

boyuan-chen committed:
Add THE_GRAPH_API_KEY

sk-enya committed:
Merge pull request #8 from bobanetwork/sahil/anchorage-withdrawal-graphoutsource

Bundling Fixes and updated env key param name!

boyuan-chen committed:
Support FDG

boyuan-chen committed:
Update tests.yml

boyuan-chen committed:
Do tests on any PR

boyuan-chen committed:
Fix api key

boyuan-chen committed:
Merge branch 'wsdt/graph-outsource' into support-fdg

boyuan-chen committed:
Update FDG functions

sk-enya commented on src/graphql.service.ts:
```diff
@@ -5,136 +5,157 @@ import {
+                gql: 'https://graphql.testnet.bnb.boba.network/subgraphs/name/boba-bnb-l2-testnet/',
                 local: 'http://127.0.0.1:8002/subgraphs/name/boba/Bridges',
             },
+            [EGraphQLService.AnchorageBridge]: {
+                gql: 'https://graphql.bnb.boba.network/subgraphs/name/anchorage-bridging-boba-bnb',
+            },
```
@wsdt looks like subgraphs is refered as mainnet one like Boba BNB above.

https://github.com/bobanetwork/graphql-utils/pull/6/files#diff-598c468d2a6aebfb01813d5057757685113469de2752fb80cadc288859448389R73-R75

wsdt replied:
@sk-enya what do you mean with is referred as mainnet? It is within the boba bnb mainnet config

sk-enya replied:
I mean this url is for BOBA bnb mainnet subgraph.!

sk-enya committed:
fix add proven timestamp

sk-enya committed:
Merge pull request #10 from bobanetwork/fix/added-proven-timestamp

fix add proven timestamp

sk-enya committed:
Merge pull request #9 from bobanetwork/support-fdg

Support FDG

boyuan-chen committed:
Update boba bnb testnet endpoint

sk-enya committed:
change:
- updated queries for bnb network to fetch initiated eth or erc20 tokens.
- updated messaged passed with different configuration for bnb using address.
- refactored the function to fetch content based on active bnb network.

sk-enya committed:
change to list the withdrawal transaction for bnb testnet

sk-enya committed:
removed console

sk-enya committed:
change: console cleanup

sk-enya committed:
updated spec for withdrawal initiateds bnb

sk-enya committed:
Merge pull request #11 from bobanetwork/sahil/subgraph-changes-bnb-withdrawal

Subgraph fetch for withdrawal trx for BNB Testnet
